resources/migration-guide-v2-v5.md

```diff
@@ -0,0 +1,402 @@
+- The new class is located in the [request.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/rpc/request.ts#L542) file.
+
+#### Migration Example
+
+#### Old Syntax (2.x):
+
+```tsx
+import { PurseIdentifier } from 'casper-js-sdk';
+
+const balanceByPublicKey = await client.queryBalance(
+  PurseIdentifier.MainPurseUnderPublicKey,
+  privateKey.publicKey.toHex(false)
+);
+
+const balanceByAccountHash = await client.queryBalance(
+  PurseIdentifier.MainPurseUnderAccountHash,
+  privateKey.publicKey.toAccountHashStr()
+);
+```
+
+#### New Syntax (5.x):
+
+```tsx
+import {
+  PurseIdentifier,
+  RpcClient,
+  HttpHandler
+} from 'casper-js-sdk';
+
+const rpcHandler = new HttpHandler('http://<Node Address>:7777/rpc');
+const rpcClient = new RpcClient(rpcHandler);
+
+try {
+  const balanceByPublicKey = await rpcClient.queryLatestBalance(
+    PurseIdentifier.fromPublicKey(privateKey.publicKey.toHex(false))
+  );
+
+  const balanceByAccountHash = await rpcClient.queryLatestBalance(
+    PurseIdentifier.fromAccountHash(
+      privateKey.publicKey.accountHash()
+    )
+  );
+} catch (e) {}
+```
+
+Explore the new PurseIdentifier class [here](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/rpc/request.ts#L542).
+
+---
+
+### 7. **`DeployUtil` to `Deploy`**
+
+The `DeployUtil` module in the previous versions of casper-js-sdk has been replaced with the `Deploy` class in version 5.x. Functionality for deploy creation `DeployUtil` become [Deploy](../src/types/Deploy.ts) class with static methods. [See how to create Deploy here](../README.md#creating-a-legacy-deploy-1)
+
+### Key Differences
+
+1. **Module Structure**:
+   - The old DeployUtil module had multiple static utility methods.
+   - The new Deploy class organizes these functionalities as methods on a single object.
+2. **Serialization and Deserialization**:
+   - The new module uses TypedJSON for serialization/deserialization, ensuring type safety.
+3. **Improved Method Organization**:
+   - Methods are grouped logically, providing a cleaner and more intuitive API.
+
+### Additional Notes
+
+- The new Deploy class provides a more type-safe and modular approach, reducing errors and improving code readability.
+- Refer to the [Deploy Class Documentation](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/types/Deploy.ts) for a complete list of available methods.
+- Ensure that you update your dependencies and verify compatibility with the 5.x SDK before migrating.
+
+---
+
+### 8. **`Contracts` Abstraction Removed**
+
+In the newer version of `casper-js-sdk`, we’ve removed the abstraction previously used for interacting with smart contracts. To provide greater flexibility and clarity, we have introduced two distinct classes: `ContractHash` and `ContractPackageHash`. These allow developers to directly install and call smart contracts with greater control. Below are two popular examples to help you transition smoothly:
```

Comment:
When I read this paragraph, I get the impression that the new `ContractHash` and `ContractPackageHash` classes are the key to build contract calls. Then, I see the examples, and that's not the case. The examples build a `Deploy` where the contract install or call are managed by a session in the form of an `ExecutableDeployItem` instance. This seems to be the important change from v2.